tensorflow/python/ops/bincount_ops.py

```diff
@@ -215,6 +217,12 @@ def bincount(arr,
           binary_output=binary_output)
     else:
       weights = validate_dense_weights(arr, weights, dtype)
+      # check if we are using DenseBincount with GPU config
+      if "GPU" in set([d.device_type for d in tf_config.list_physical_devices()]):
```

Comment:
It's not generally useful to put in a test like this - it's not important that the GPU version is unimplemented.  Instead, we should just skip the test for GPU.

You can use something like
```
if test_util.is_gpu_available():
  self.skipTest("GPU kernel does not support weights.")
```

Reply:
This line isn't a test, its to catch the error when `DenseBincount` is used with a GPU config. Is it still appropriate to use `test_util.is_gpu_available()` in this case?

Reply:
Ah, right, sorry.  I read this PR from bottom up and assumed this was a test.

Unimplemented errors should go in the kernel implementation itself.  It's not an inherent issue with the op.  For example, it might work just fine with XLA, or oneDNN, or tensorflow-metal, or whatever other plugin/backend people use.  It's only the built-in TF kernel that fails - and I believe it already throws an unimplemented error?